fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/FieldValue.java

```diff
@@ -320,13 +324,57 @@ public static Optional<FieldPath> stripFieldPrefixMaybe(@Nonnull FieldPath field
         final var potentialPrefixFieldOrdinals = potentialPrefixPath.getFieldOrdinals();
         final var potentialPrefixFieldTypes = potentialPrefixPath.getFieldTypes();
         for (int i = 0; i < potentialPrefixPath.size(); i++) {
-            if (fieldPathOrdinals.get(i) != potentialPrefixFieldOrdinals.get(i) || !fieldPathTypes.get(i).equals(potentialPrefixFieldTypes.get(i))) {
+            if (fieldPathOrdinals.get(i) != potentialPrefixFieldOrdinals.get(i) || !equalIgnoringFieldNames(fieldPathTypes.get(i), potentialPrefixFieldTypes.get(i), new HashSet<>())) {
                 return Optional.empty();
             }
         }
         return Optional.of(fieldPath.subList(potentialPrefixPath.size(), fieldPath.size()));
     }
 
+    @SuppressWarnings("PMD.CompareObjectsWithEquals") // Deliberate use of reference equality
+    private static boolean equalIgnoringFieldNames(@Nonnull Type t1, @Nonnull Type t2, @Nonnull Set<Pair<Type, Type>> explored) {
+        if (t1 == t2) {
+            // Short circuit for reference equality
+            return true;
+        }
+        if (t1.isNullable() != t2.isNullable()) {
+            return false;
+        }
+        if (t1.isArray()) {
+            if (!t2.isArray()) {
+                return false;
+            }
+            @Nullable Type t1Element = ((Type.Array)t1).getElementType();
```

Comment:
This can only be `null` if the type is erased which should not happen as these types are resolved. I am assuming you are being defensive here and just return `false` if it happens. Maybe you can do a `Objects.requireNonNull()` to enforce it is not `null`. Or alternatively, maybe we can make the API better. There are other places where we have different flavors, life e.g. `getThing()` and `getThingMaybe()` where the latter returns an `Optional` while the former throws if it is empty.

Reply:
Now irrelevant as this method has been removed